kratos/geometries/brep_surface.h

```diff
@@ -196,13 +199,97 @@ class BrepSurface
         return typename BaseType::Pointer( new BrepSurface( ThisPoints ) );
     }
 
+    ///@}
+    ///@name Access to Geometry Parts
+    ///@{
+
+    /**
+    * @brief This function returns the pointer of the geometry
+    *        which is corresponding to the trim index.
+    *        Surface of the geometry is accessable with SURFACE_INDEX.
+    * @param Index: trim_index or SURFACE_INDEX.
+    * @return pointer of geometry, corresponding to the index.
+    */
+    GeometryPointer pGetGeometryPart(IndexType Index) override
+    {
+        const auto& const_this = *this;
+        return std::const_pointer_cast<GeometryType>(
```

Comment:
hm I am not sure abt this ...

Reply:
It's working. And so far it reduces the code significantly. I'll keep an eye on it to see how it performs in the future. If it shows off as bad we'll just duplicate the things...